src/study.js

```diff
@@ -125,15 +133,18 @@ class Study {
       }
 
       if (shouldRemoveBucket) {
-        if (!isServer) { document.body.classList.remove(`${test.name}--${oldAssignments[test.name]}`); }
+        this.classList[test.name] = null;
         delete this.userAssignments[test.name];
       } else {
         // Add to our assignments
-        if (!isServer) { document.body.classList.add(`${test.name}--${bucket}`); }
+        this.classList[test.name] = bucket;
         this.userAssignments[test.name] = bucket;
       }
     });
 
+    if (!isServer && document.body) applyClasses();
+    else document.addEventListener('DOMContentLoaded', applyClasses);
```

Comment:
What @yowainwright said. Is this fully supported across the browsers we support?